pallets/creditcoin/src/helpers/external_address.rs

```diff
@@ -53,160 +51,24 @@ fn btc_address_is_well_formed(address: &[u8]) -> bool {
 
 // ether-like
 
-#[derive(Copy, Clone)]
-enum AsciiCase {
-	None,
-	Lower,
-	Upper,
-}
-
-const HEX_PREFIX: &[u8] = b"0x";
-const ETH_ADDRESS_LENGTH: usize = 42;
-const ETH_ADDRESS_VALUE_LENGTH: usize = ETH_ADDRESS_LENGTH - HEX_PREFIX.len();
+const ETH_ADDRESS_LENGTH: usize = 20;
 
 fn eth_address_is_well_formed(address: &[u8]) -> bool {
-	if !address.starts_with(HEX_PREFIX) || address.len() != ETH_ADDRESS_LENGTH {
-		return false;
-	}
-
-	let address_value = &address[HEX_PREFIX.len()..];
-
-	if eth_address_is_checksummed(address_value) {
-		eth_address_checksum_valid(address_value)
-	} else {
-		true
```

Comment:
Jeremy and I discussed this and decided this was acceptable, we can do the checksum on the client-side. PolkadotJS turns `0x12345` into [1, 2, 3, 4, 5] (instead of treating it as a string) so we only receive the bytes - this was the cause of the integration test failures. Since we only receive the raw bytes we can't perform the checksum.